Analysis/src/AnaTuple.cpp

```diff
@@ -92,20 +93,52 @@ void AnaTupleWriter::AddEvent(EventInfo& event, const DataIdMap& dataIds, const
         tuple().all_mva_scores.push_back(static_cast<float>(mva_score));
     }
 
-    auto fill_unc_weight_vec = [](const std::map<UncertaintyScale, float>& weights_in, std::vector<float>& weights_out) {
+    auto fill_unc_weight_vec = [](const std::map<UncertaintyScale, float>& weights_in, std::vector<float>& weights_out,
+                                  bool store_relative = true) {
+        static const std::vector<UncertaintyScale> scales = { UncertaintyScale::Up, UncertaintyScale::Down };
+
         weights_out.clear();
-        weights_out.push_back(weights_in.at(UncertaintyScale::Central));
-        if(weights_in.count(UncertaintyScale::Up))
-            weights_out.push_back(weights_in.at(UncertaintyScale::Up));
-        if(weights_in.count(UncertaintyScale::Down))
-            weights_out.push_back(weights_in.at(UncertaintyScale::Down));
+        const float central_w = weights_in.at(UncertaintyScale::Central);
+        if(!store_relative)
+            weights_out.push_back(central_w);
+        for(auto scale : scales) {
+            if(!weights_in.count(scale)) break;
+            float w = weights_in.at(scale);
+            if(store_relative)
+                w /= central_w;
+            weights_out.push_back(w);
+        }
     };
 
     if(!event->isData){
-        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Loose), tuple().btag_weight_Loose);
-        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Medium), tuple().btag_weight_Medium);
-        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Tight), tuple().btag_weight_Tight);
+        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Loose), tuple().btag_weight_Loose, false);
+        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Medium), tuple().btag_weight_Medium, false);
+        fill_unc_weight_vec(btag_weights.at(DiscriminatorWP::Tight), tuple().btag_weight_Tight, false);
+        fill_unc_weight_vec(uncs_weight_map.at(UncertaintySource::EleTriggerUnc), tuple().unc_EleTriggerUnc, true);
```

Comment:
all `fill_unc_weight_vec` except the ones for `btag_weights` should be called inside `if(even.GetEventCandidate (). GetUncSource() == UncertaintySource::None)`